napari/plugins/_npe2.py

```diff
@@ -182,6 +182,32 @@ def _fmt_exts(es):
     )
 
 
+@npe2_or_return({})
+def get_readers(path: str) -> Dict[str, str]:
+    """Get valid reader display_name: plugin_name mapping given path.
+
+    Iterate through compatible readers for the given path and return
+    dictionary of display_name to plugin_name for each reader
+
+    Parameters
+    ----------
+    path : str
+        path for which to find compatible readers
+
+    Returns
+    -------
+    Dict[str, str]
+        Dictionary of display_name to plugin_name
+    """
+    pm = npe2.PluginManager.instance()
+    readers = list(pm.iter_compatible_readers(path))
+    reader_plugin_names = {
+        pm.get_manifest(reader.command).display_name: reader.plugin_name
+        for reader in readers
+    }
```

Comment:
```suggestion
    reader_plugin_names = {
        pm.get_manifest(reader.command).display_name: reader.plugin_name
        for reader in pm.iter_compatible_readers(path)
    }
```

(ever so slightly more performant, no need to cast a generator to a list and then re-iterate, if you're just going to iterate through it once)